apollo-router/src/spec/query/change.rs

```diff
@@ -132,58 +165,79 @@ impl<'a> QueryHashVisitor<'a> {
                 b.hash(self);
             }
             schema::Value::List(l) => {
-                "list[".hash(self);
+                "^list[".hash(self);
                 for v in l.iter() {
                     self.hash_value(v);
                 }
-                "]".hash(self);
+                "^]".hash(self);
             }
             schema::Value::Object(o) => {
-                "object{".hash(self);
+                "^object{".hash(self);
                 for (k, v) in o.iter() {
+                    "^key".hash(self);
+
                     k.hash(self);
-                    ":".hash(self);
+                    "^:".hash(self);
                     self.hash_value(v);
                 }
                 "}".hash(self);
```

Comment:
```suggestion
                "^}".hash(self);
```

For consistency, since we use `^]` to end the list.